system/services/assetManager/AssetManagerService.cfc

```diff
@@ -210,13 +211,16 @@ component displayName="AssetManager Service" {
 		  required string  type
 		, required string  size
 		, required string  folderId
+		,          string  assetId
 		,          string  currentFolderId = ""
 		,          boolean throwIfNot   = false
 		,          struct  restrictions = getFolderRestrictions( arguments.folderId )
 	) {
 		var typeDisallowed = restrictions.allowedExtensions.len() && !ListFindNoCase( restrictions.allowedExtensions, "." & arguments.type );
 		var sizeInMb       = arguments.size / 1048576;
 		var tooBig         = restrictions.maxFileSize && sizeInMb > restrictions.maxFileSize;
+		var fileExist      = _getAssetDao().dataExists( filter = { id = arguments.assetId } );
```

Comment:
What we should be checking for here is that _another_ asset does not already exist in the new folder that has the same _name_ as the current asset.

If we just check for the existance of the asset, then we will never be able to move assets between folders.